packages/block-library/src/video/edit.js

```diff
@@ -74,8 +79,28 @@ class VideoEdit extends Component {
 		this.setState( { editing: false } );
 	}
 
+	onSelectPoster( image ) {
+		const { setAttributes, clientId } = this.props;
+		setAttributes( { poster: image.url } );
+		document.getElementById( 'block-' + clientId ).getElementsByTagName( 'video' )[ 0 ].load();
```

Comment:
I think we should use `ref` in this case. See the related docs: https://reactjs.org/docs/refs-and-the-dom.html

Usage example in the existing code:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/block/edit-panel/index.js#L15
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/block/edit-panel/index.js#L66
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/block/edit-panel/index.js#L35

Reply:
By the way, this logic should probably be moved to `componenDidUpdate` lifecycle method as it should be triggered before `this.props.attributes.poster` isn't updated.